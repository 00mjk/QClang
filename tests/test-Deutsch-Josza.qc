int main() {
    /* QClang implementation of Deutsch-Josza Algorith */
    int i;
    int strlen;

    qubit[] test_bits;
    bit[] measure_bits;
    qubit answer;

    /* Create Qubit array for oracle query */
    strlen = 10;
    test_bits = new qubit[](strlen);
    measure_bits = new bit[](strlen);

    /* Create superposition state */
    test_bits = hadamard(test_bits);

    answer = false;
    answer = hadamard(answer);

    /* Query oracle  */
    answer = !answer;

    /* Apply hadamard again */
    test_bits = hadamard(test_bits);

    /* Measure */
    measure_bits = measure(test_bits);

    return 0;
}
