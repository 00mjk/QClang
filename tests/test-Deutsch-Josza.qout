COMPLETED
{'0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0': 1}
dict_keys(['0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0'])
