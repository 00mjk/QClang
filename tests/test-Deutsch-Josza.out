OPENQASM 2.0;
include "qelib1.inc";
qreg q[1];
creg c[1];
h q;
qreg tmp2_qt0[1];
qreg answer_q0[1];
qreg constemp0_qt1[1];
qreg constemp1_qt2[1];
qreg constemp2_qt3[1];
qreg constemp3_qt4[1];
qreg constemp4_qt5[1];
qreg constemp5_qt6[1];
qreg constemp6_qt7[1];
qreg constemp7_qt8[1];
qreg constemp8_qt9[1];
qreg constemp9_qt10[1];
creg constemp10_bt11[1];
creg constemp11_bt12[1];
creg constemp12_bt13[1];
creg constemp13_bt14[1];
creg constemp14_bt15[1];
creg constemp15_bt16[1];
creg constemp16_bt17[1];
creg constemp17_bt18[1];
creg constemp18_bt19[1];
creg constemp19_bt20[1];
qreg constemp20_qt21[1];
qreg constemp21_qt22[1];
qreg constemp22_qt23[1];
qreg constemp23_qt24[1];
qreg constemp24_qt25[1];
qreg constemp25_qt26[1];
qreg constemp26_qt27[1];
qreg constemp27_qt28[1];
qreg constemp28_qt29[1];
qreg constemp29_qt30[1];
h constemp9_qt10;
h constemp8_qt9;
h constemp7_qt8;
h constemp6_qt7;
h constemp5_qt6;
h constemp4_qt5;
h constemp3_qt4;
h constemp2_qt3;
h constemp1_qt2;
h constemp0_qt1;
qreg temp_30[1];
h temp_30;
x temp_30[0];
qreg constemp31_qt32[1];
qreg constemp32_qt33[1];
qreg constemp33_qt34[1];
qreg constemp34_qt35[1];
qreg constemp35_qt36[1];
qreg constemp36_qt37[1];
qreg constemp37_qt38[1];
qreg constemp38_qt39[1];
qreg constemp39_qt40[1];
qreg constemp40_qt41[1];
h constemp9_qt10;
h constemp8_qt9;
h constemp7_qt8;
h constemp6_qt7;
h constemp5_qt6;
h constemp4_qt5;
h constemp3_qt4;
h constemp2_qt3;
h constemp1_qt2;
h constemp0_qt1;
creg constemp41_bt42[1];
creg constemp42_bt43[1];
creg constemp43_bt44[1];
creg constemp44_bt45[1];
creg constemp45_bt46[1];
creg constemp46_bt47[1];
creg constemp47_bt48[1];
creg constemp48_bt49[1];
creg constemp49_bt50[1];
creg constemp50_bt51[1];
creg constemp9_qt10_mb51[1];
measure constemp9_qt10 -> constemp9_qt10_mb51;
creg constemp8_qt9_mb52[1];
measure constemp8_qt9 -> constemp8_qt9_mb52;
creg constemp7_qt8_mb53[1];
measure constemp7_qt8 -> constemp7_qt8_mb53;
creg constemp6_qt7_mb54[1];
measure constemp6_qt7 -> constemp6_qt7_mb54;
creg constemp5_qt6_mb55[1];
measure constemp5_qt6 -> constemp5_qt6_mb55;
creg constemp4_qt5_mb56[1];
measure constemp4_qt5 -> constemp4_qt5_mb56;
creg constemp3_qt4_mb57[1];
measure constemp3_qt4 -> constemp3_qt4_mb57;
creg constemp2_qt3_mb58[1];
measure constemp2_qt3 -> constemp2_qt3_mb58;
creg constemp1_qt2_mb59[1];
measure constemp1_qt2 -> constemp1_qt2_mb59;
creg constemp0_qt1_mb60[1];
measure constemp0_qt1 -> constemp0_qt1_mb60;
